framework/src/vectorpostprocessors/VectorPostprocessorData.C

```diff
@@ -29,38 +29,57 @@ VectorPostprocessorData::hasVectorPostprocessor(const std::string & name)
 VectorPostprocessorValue &
 VectorPostprocessorData::getVectorPostprocessorValue(const VectorPostprocessorName & vpp_name, const std::string & vector_name)
 {
-  VectorPostprocessorValue * & pp_val = _values[vpp_name][vector_name];
+  _requested_items.emplace(vpp_name + "::" + vector_name);
 
-  if (pp_val == NULL)
-    pp_val = &declareRestartableDataWithObjectName<VectorPostprocessorValue>(vpp_name + "_" + vector_name, "values");
-
-  return *pp_val;
+  return getVectorPostprocessorHelper(vpp_name, vector_name, true);
 }
 
 VectorPostprocessorValue &
 VectorPostprocessorData::getVectorPostprocessorValueOld(const VectorPostprocessorName & vpp_name, const std::string & vector_name)
 {
-  VectorPostprocessorValue * & pp_val = _values_old[vpp_name][vector_name];
-
-  if (pp_val == NULL)
-    pp_val = &declareRestartableDataWithObjectName<VectorPostprocessorValue>(vpp_name + "_" + vector_name, "values_old");
+  _requested_items.emplace(vpp_name + "::" + vector_name);
```

Comment:
This means that you will create a dependency on this VPP... even though it's only a dependency on the _old_ value of this VPP.

That can lead to erroneous cyclic dependencies (like the kind we've had to clear up elsewhere).

I think that for now it's ok to just _not_ note the dependency on the old value.  It should always be available...